esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1205,14 +1276,19 @@ EspAuthState CEspHttpServer::authExistingSession(EspAuthRequest& authReq, unsign
 
     VStringBuffer xpath("%s[@port=\"%d\"]/%s[%s='%u']", PathSessionApplication, authReq.authBinding->getPort(), PathSessionSession, PropSessionID, sessionID);
     IPropertyTree* sessionTree = espSessions->queryBranch(xpath.str());
-    if (!authReq.serviceName.isEmpty() && !authReq.methodName.isEmpty() &&
-        strieq(authReq.serviceName.str(), "esp") && strieq(authReq.methodName.str(), "get_session_timeout"))
+    if (!authReq.serviceName.isEmpty() && !authReq.methodName.isEmpty() && strieq(authReq.serviceName.str(), "esp"))
     {
         StringBuffer content;
         ESPSerializationFormat respFormat = m_request->queryContext()->getResponseFormat();
-        createGetSessionTimeoutResponse(content, respFormat, sessionTree);
-        sendMessage(content.str(), (respFormat == ESPSerializationJSON) ? "application/json" : "text/xml");
-        return authTaskDone;
+        if (strieq(authReq.methodName.str(), "get_session_timeout"))
+            createGetSessionTimeoutResponse(content, respFormat, sessionTree);
+        else if (strieq(authReq.methodName.str(), "reset_session_timeout"))
+            resetSessionTimeout(authReq, sessionID, content, respFormat, sessionTree);;
+        if (!content.isEmpty())
+        {
+            sendMessage(content.str(), (respFormat == ESPSerializationJSON) ? "application/json" : "text/xml");
+            return authTaskDone;
```

Comment:
The "return authTaskDone" was not unconditional. It happens only if the method name is "get_session_timeout". Now, it happends only if the method name is "get_session_timeout" or "reset_session_timout".